### PR TITLE
twig2 compatibility

### PR DESCRIPTION
**Pull request by toooni**
makes use of php-translation/extractor twig2 support

### CODE REVIEW

File: Resources/config/extractors.yml [hunk 1]
```diff
@@ -53,11 +53,11 @@ services:
 
   # Twig Visitors:
   php_translation.extractor.twig.visitor.translation_block:
-    class: Translation\Extractor\Visitor\Twig\TranslationBlock
+    class: %php_translation.extractor.twig.visitor.translation_block.class%
```
Comment (Nyholm) [inline Resources/config/extractors.yml]: Make sure to quote this parameter

File: Resources/config/extractors.yml [hunk 2]
```diff
@@ -53,11 +53,11 @@ services:
...
     tags:
       - { name: 'php_translation.visitor', type: 'twig' }
 
   php_translation.extractor.twig.visitor.translation_filter:
-    class: Translation\Extractor\Visitor\Twig\TranslationFilter
+    class: %php_translation.extractor.twig.visitor.translation_filter.class%
```
Comment (Nyholm) [inline Resources/config/extractors.yml]: Make sure to quote this parameter

**Comment by toooni** [comment]
@Nyholm Having two separate files could lead to problems with autocompletion (PHPStorm might lead to the wrong file)? But I am not sure what is better.
You can tell me what to do. It's not very hard to change the PR.

**Comment by Nyholm** [comment]
PHPStorm does not use Yaml files to autocomplete services. It is using the `var/cache/dev/appDevDebugProjectContainer.php`. That file is built on cache warmup. So we will not have any issues with multiple files. 

There is no big deal and we could change this in a patch release later if we want to.

File: DependencyInjection/TranslationExtension.php [hunk 3]
```diff
@@ -44,6 +48,14 @@ public function load(array $configs, ContainerBuilder $container)
         $container->getDefinition('php_translation.extractor.php.visitor.FormTypeChoices')
             ->addMethodCall('setSymfonyMajorVersion', [Kernel::MAJOR_VERSION]);
 
+        if (\Twig_Environment::MAJOR_VERSION === 1) {
```
Comment (Nyholm) [inline DependencyInjection/TranslationExtension.php]: This variable was introduced in 1.28.0. We should use `version_compare` and `\Twig_Environment::VERSION` instead.

Comment (toooni) [inline DependencyInjection/TranslationExtension.php]: @Nyholm done

**Comment by toooni** [comment]
@Nyholm you're right about the autocompletion. Another thing then would be the duplication of all other services in `extractors.yml`. A solution would be splitting `extractors.yml` into `extractory_twig.yml` and `extractors.yml` or something.

**Comment by Nyholm** [comment]
Yes, I agree.. 
Let's solve that later =) I think this is good for now.

**Pull request merged**